build.rs

```diff
@@ -0,0 +1,13 @@
+fn main() {
+    let is_nightly = version_check::is_feature_flaggable() == Some(true);
+    let is_at_least_1_34 = version_check::is_min_version("1.34.0").unwrap_or(false);
+    let is_at_least_1_51 = version_check::is_min_version("1.51.0").unwrap_or(false);
+    
+    if !is_at_least_1_34 && !is_nightly {
+        println!("cargo:warning=itertools requires rustc => 1.34.0");
+    }
+
+    if is_at_least_1_51 || is_nightly {
+        println!("cargo:rustc-cfg=has_min_const_generics");
+    }
+}
```

Comment:
Usually, I like the idea of having everything automated, but I am  not sure if we should go with a `build.rs` and an additional dependency. My first idea was to use a feature flag (that would probably be off by default) that the user can enable if desired.

Reply:
> My first idea was to use a feature flag (that would probably be off by default) that the user can enable if desired.

I think feature flags to enable things that are already available in the latest stable Rust and have no further compile time or dependency drawbacks makes no sense.

1. You could end up with dozens of feature flags for minor features, or hold back progress because a feature that would otherwise be merged would now be too minor to accept due to introducing a new feature flag.
2. You end up with useless features that you can't remove without a breaking change as your MSRV goes up.
3. It's un-ergonomic as it adds an extra step for the user.

These drawbacks while it could be done completely and correctly automatically are unacceptable in my opinion. If you are hesitant regarding the `version-check` dependency, I'd just like to note that it's tiny, has no further downstream dependencies, and is already relied on by crates such as `time`, `nom`, `rocket`, `fd-find` among others.

Reply:
@Philippe-Cholet, @phimuemue, perhaps it's time we updated our MSRV to 1.51 (which is two years old at this point). 

While I don't mind the version-detection approach, I would like us to adopt it in tandem with changes to our CI that ensure we are testing on all detected versions. I'd also like to perhaps avoid taking the dependency on `rust_version`. This would all be a substantial change, and outside the scope of this PR.

My vote is that we increase our MSRV. We can aways *decrease* it in the future.

Reply:
@jswrenn I sure don't mind increasing the MSRV but I would suggest we release the 0.13.0 first, and then increase the MSRV in 0.14.0 to not require the build script (in which case **orlp** will have enough time to work on this).